### ISSUE TITLE
flutter build apk builds flx twice

### ISSUE THREAD

**Issue by jensjoha** [flutter/flutter#15612]
`flutter clean ; time flutter build apk --debug --preview-dart-2`

internally runs `build flx` twice (https://github.com/flutter/flutter/blob/master/packages/flutter_tools/gradle/flutter.gradle#L297).

Instrumenting the code so it prints when it builds flx makes it obvious; also removing the double-call makes stuff faster:

Without two calls:
```
flutter/examples/flutter_gallery$ flutter clean ; time flutter build apk --debug --preview-dart-2
Building flutter tool...
Deleting 'build/'.
Running "flutter packages get" in flutter_gallery...         0.5s
Initializing gradle...                                       0.6s
Resolving dependencies...                                    0.6s
Running 'gradlew assembleDebug'...                               
BUILDING FLX!!
Built build/app/outputs/apk/debug/app-debug.apk (37.3MB).

real	0m15.731s
user	0m4.856s
sys	0m0.592s
```

With two calls (i.e. as it currently is):
```
flutter/examples/flutter_gallery$ flutter clean ; time flutter build apk --debug --preview-dart-2
Deleting 'build/'.
Initializing gradle...                                       0.5s
Resolving dependencies...                                    0.6s
Running 'gradlew assembleDebug'...                               
BUILDING FLX!!
BUILDING FLX!!
Built build/app/outputs/apk/debug/app-debug.apk (37.3MB).

real	0m18.106s
user	0m3.780s
sys	0m0.508s
```

/cc @mraleph @mravn-google @szakarias

**Comment by szakarias** [comment]
Thanks, @jensjoha. There has earlier been a reason for building twice, but it is not necessary anymore. I'll make sure to remove the extra build step.

**Comment by szakarias** [comment]
Fixed in #16416
